tests/test_tools.py

```diff
@@ -310,7 +315,7 @@ def test_subexpr_from_unbalanced_parens():
         ]
     for expr, exp in cases:
         obs = subexpr_from_unbalanced(expr, '(', ')')
-        yield assert_equal, exp, obs
+        assert exp == obs
```

Comment:
Tests like these should be parameterizations, since all cases are independent of one another and a failure in one should not prevent the other tests from running.